### PR TITLE
devicetree: Remove DT_SRAM_{BASE_ADDRESS,SIZE}, use CONFIG_* versions

### PR DESCRIPTION
**Pull request by ulfalizer**
The SRAM address and size are currently available as both
DT_SRAM_{BASE_ADDRESS,SIZE} and as CONFIG_SRAM_{BASE_ADDRESS,SIZE} (via
the Kconfig preprocessor).

Use the CONFIG_SRAM_* versions everywhere, and remove generation of the
DT_SRAM_* versions from gen_defines.py.

The Kconfig symbols currently depend on 'ARC || ARM || NIOS2 || X86'.
Not sure why, so I removed it.

It looks like no configuration files set CONFIG_SRAM_* at the moment, so
another option might be to use the DT_* symbols everywhere instead. Some
Kconfig.defconfig.series files add defaults to them though.

Also improve the help texts for CONFIG_SRAM_* to say that they normally
come from devicetree rather than configuration files.

### CODE REVIEW

**Review by dbkinder — APPROVED** [review]
no rst files changes.  Kconfig doc changes LGTM.

**Review by nashif — APPROVED** [review]

File: soc/xtensa/intel_s1000/dts_fixup.h [hunk 1]
```diff
@@ -12,8 +12,8 @@
 #define DT_UART_NS16550_PORT_0_IRQ_FLAGS	DT_NS16550_80800_IRQ_0_SENSE
 #define DT_UART_NS16550_PORT_0_CLK_FREQ	DT_NS16550_80800_CLOCK_FREQUENCY
 
-#define DT_L2_SRAM_BASE				DT_SRAM_BASE_ADDRESS
-#define DT_L2_SRAM_SIZE				DT_SRAM_SIZE * 1024
+#define DT_L2_SRAM_BASE				CONFIG_SRAM_BASE_ADDRESS
+#define DT_L2_SRAM_SIZE				CONFIG_SRAM_SIZE * 1024
```
Comment (ydamigos) [inline soc/xtensa/intel_s1000/dts_fixup.h]: Personally, I don't like using CONFIG_* macros in `dts_fixup.h` files. It feels strange.

**Review by ydamigos — COMMENTED** [review]
Should we use DT_SRAM_* instead of CONFIG_SRAM_*? As far as I know, we are using DT to describe hardware. Did something change?

**Comment by ulfalizer** [comment]
> Should we use DT_SRAM_* instead of CONFIG_SRAM_*? As far as I know, we are using DT to describe hardware. Did something change?

Mentioned it in the commit message:

    It looks like no configuration files set CONFIG_SRAM_* at the moment, so another
    option might be to use the DT_* symbols everywhere instead. Some
    Kconfig.defconfig.series files add defaults to them though.

Getting rid of `CONFIG_SRAM_*` later would be nice. It would break compatibility with any external projects that set them, but maybe no one does, and it's probably easy to fix.

While they're there, it's nice to have things consistent at least.

**Pull request merged**